exporters/write_buffer.py

```diff
@@ -155,20 +161,24 @@ def get_current_buffer_file_for_group(self, key):
         return buffer_file
 
 
-class WriteBuffer(object):
+class WriteBuffer(BasePipelineItem):
+    """
+    This module handles buffering during batches writing.
+    """
+    supported_options = {
+        'items_per_buffer_write': {'type': six.integer_types, 'default': ITEMS_PER_BUFFER_WRITE},
+        'size_per_buffer_write': {'type': six.integer_types, 'default': SIZE_PER_BUFFER_WRITE},
```

Comment:
these options must be writer options, regardless they are passed to the write buffer. As explained in the other comment, moving out from writer options will break all existing configuration files. Also because it will complicate configuration files, adding a new level of configuration for very commonly used options.

Lets avoid to have to define write buffer options if not really needed. When I suggested to add options to write buffer I was thinking on other extra tunning parameters, not these ones. You must leave WriteBuffer supported_options attribute as an empty dict here. It will be there for subclasses only.

Also, with this approach this PR will be much smaller than what you did (you did too many unneeded changes)